default.nix

```diff
@@ -21,10 +21,22 @@ with stdenv.lib; let
       path = fetchurl { name = fname; url = resolved; "${hashType}" = hash; };
     };
 
+  overrideTgz = src: runCommand "${src.name}.tgz" {} ''
+    cp -r --reflink=auto ${src} ./package
+    chmod +w ./package ./package/package.json
+    # scripts are not supported
+    ${jq}/bin/jq '.scripts={}' ${src}/package.json > ./package/package.json
+    tar --sort=name --owner=0:0 --group=0:0 --mtime='UTC 2019-01-01' -czf $out package
+  '';
+
+  overrideToFetch = pkg: { path = "${overrideTgz pkg}"; };
+
   depToFetch = args @ { resolved ? null, dependencies ? {}, ... }:
     (optional (resolved != null) (depFetchOwn args)) ++ (depsToFetches dependencies);
 
-  cacheInput = oFile: iFile: writeText oFile (toJSON (listToAttrs (depToFetch iFile)));
+  cacheInput = oFile: iFile: overrides:
```

Comment:
`cacheInput` is also called in `mkYarnPackage`, you should pass `overrides` parameter there as well (you can just pass an empty attrset if you don't want to bother adding override support for yarn)